server.py

```diff
@@ -113,12 +114,34 @@ def index():
     ), status.HTTP_200_OK
 
 @app.route(PATH_INVENTORY, methods=[GET])
-def list_all_prod_info():
-    """ Return all entries in the Inventory system """
-    all_prod_info = ProductInformation.list_all()
+def query_prod_info():
+    """ Query specific entries in the Inventory system """
+    all_prod_info = []
+    if request.args.get('prod_name'):
+        prod_name = request.args.get('prod_name')
+        if prod_name == None:
+            abort(status.HTTP_400_BAD_REQUEST, INVALID_PARAMETER_MSG)
+        all_prod_info = ProductInformation.find_by_name(prod_name)
+    elif request.args.get('quantity'):
+        quantity = request.args.get('quantity')
+        try:
+            quantity = int(quantity)
```

Comment:
Suggest using `if isinstance(quantity, int):`

Reply:
Also, check for negative quantity request. Add tests correspondingly.

Reply:
I think the args passed into the request are all strings by default?

Reply:
I'm not sure if negative quantity is not allowed. My implementation could handle the negative quantity properly (i.e. returning an empty list if no record matches).